code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessagingUIEventObserver.kt

```diff
@@ -0,0 +1,57 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.messaging
+
+import com.adobe.marketing.mobile.MessagingEdgeEventType
+import com.adobe.marketing.mobile.aepcomposeui.interactions.UIEvent
+import com.adobe.marketing.mobile.aepcomposeui.observers.AepUIEventObserver
+import com.adobe.marketing.mobile.aepuitemplates.AepUITemplate
+import com.adobe.marketing.mobile.aepuitemplates.SmallImageTemplate
+import com.adobe.marketing.mobile.services.Log
+
+class MessagingUiEventObserver(private val callback: ContentCardCallback?) : AepUIEventObserver {
+    companion object {
+        private const val SELF_TAG = "MessagingUiEventObserver"
+    }
+
+    override fun onEvent(event: UIEvent<*, *>) {
+        when (event) {
+            is UIEvent.Display -> {
+                Log.trace(MessagingConstants.LOG_TAG, SELF_TAG, "SmallImageUI Displayed")
+                handleSmallImageTracking(event.aepUi.getTemplate(), null, MessagingEdgeEventType.DISPLAY)
+            }
+            is UIEvent.Interact -> {
+                if (callback?.onCardClick(event.aepUi.getTemplate()) == true) {
+                    Log.trace(MessagingConstants.LOG_TAG, SELF_TAG, "SmallImageUI Click")
+                    handleSmallImageTracking(event.aepUi.getTemplate(), "clicked", MessagingEdgeEventType.INTERACT)
+                }
+            }
```

Comment:
The `Interact` event's action should be more elaborate than a String. So that we can have more properties Can you make it a set of sealed `Actions` with `Click` being one of the actions? We should also be able to provide a way pass the url/uri or something wrt to the actions. Note that these actions will be different for different interactions (swipe, long press etc)

Reply:
How about these fields for Interact?

1. aepUi
2. action of type UIAction which would have different interact enums like CLICK, SWIPE etc.
3. id: Might be useful to look up in meta data? not too sure
4. actionUrl of type String. Since every clickable element's model (eg: AepButton) has actionUrl guessing it should be easy to add here